src/providers/bitbucket.ts

```diff
@@ -0,0 +1,40 @@
+import { OAuthConfig, OAuthUserConfig } from "./oauth"
+
+interface BitbucketProfile {
+  username: string
+  account_id: string
+  display_name: string
+  links: {
+    avatar: {
+      href: string
+    }
+  }
+}
+
+export default function Bitbucket<P extends Record<string, any> = BitbucketProfile>(
+  options: OAuthUserConfig<P>
+): OAuthConfig<P> {
+  return {
+    id: "bitbucket",
+    name: "Bitbucket",
+    type: "oauth",
+    authorization: {
+      url: "https://bitbucket.org/site/oauth2/authorize",
+      params: {
+        scope: ""
+      }
+    },
+    token: "https://bitbucket.org/site/oauth2/access_token",
+    userinfo: "https://api.bitbucket.org/2.0/user",
+    profile(profile) {
+      return {
+        id: profile.account_id,
+        email: null,
```

Comment:
There is no way of retrieving this? Would be nice to match the other providers if we could get it somehow.

Reply:
The `userinfo` endpoint doesn't return the user's email. We can only retrieve it in another endpoint (`/2.0/user/emails`). Should we do the request inside the profile function?

Reply:
yes